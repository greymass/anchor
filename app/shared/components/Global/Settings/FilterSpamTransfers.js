// @flow
import React, { Component } from 'react';
import { Dropdown } from 'semantic-ui-react';
import { translate } from 'react-i18next';

class GlobalSettingsFilterSpamTransfers extends Component<Props> {
  onChange = (e, { value }) => {
    const { actions } = this.props;

    actions.setSetting('filterSpamTransfersUnder', value);
  }

  render() {
    const {
      defaultValue,
      name,
      selection,
      t
    } = this.props;

    const options = [
      { key: 'global_settings_filter_spam_transfers_0_0002', value: 0.0002, text: t('global_settings_filter_spam_transfers_0_0002') },
      { key: 'global_settings_filter_spam_transfers_0_0010', value: 0.0010, text: t('global_settings_filter_spam_transfers_0_0010') },
      { key: 'global_settings_filter_spam_transfers_0_0100', value: 0.0100, text: t('global_settings_filter_spam_transfers_0_0100') },
      { key: 'global_settings_filter_spam_transfers_0_1000', value: 0.1000, text: t('global_settings_filter_spam_transfers_0_1000') },
      { key: 'global_settings_filter_spam_transfers_do_not', value: 0.0000, text: t('global_settings_filter_spam_transfers_do_not') }
    ];

    return (
      <Dropdown
        value={defaultValue || 0.0000}
        name={name}
        onChange={this.onChange}
        options={options}
        selection={selection}
      />
    );
  }
}

export default translate('global')(GlobalSettingsFilterSpamTransfers);
