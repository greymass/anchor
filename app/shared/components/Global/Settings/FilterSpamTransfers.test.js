import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalSettingsFilterSpamTransfers } from './FilterSpamTransfers';

const mockProps = {
  defaultValue: 'default',
  name: 'transfer filter',
  selection: false,
  t: text => text,
};

describe('GlobalSettingsFilterSpamTransfers', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalSettingsFilterSpamTransfers {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
