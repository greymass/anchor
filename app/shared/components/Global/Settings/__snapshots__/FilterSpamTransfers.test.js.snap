// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalSettingsFilterSpamTransfers renders correctly 1`] = `
<Dropdown
  additionLabel="Add "
  additionPosition="top"
  closeOnBlur={true}
  deburr={false}
  icon="dropdown"
  minCharacters={1}
  name="transfer filter"
  noResultsMessage="No results found."
  onChange={[Function]}
  openOnFocus={true}
  options={
    Array [
      Object {
        "key": "global_settings_filter_spam_transfers_0_0002",
        "text": "global_settings_filter_spam_transfers_0_0002",
        "value": 0.0002,
      },
      Object {
        "key": "global_settings_filter_spam_transfers_0_0010",
        "text": "global_settings_filter_spam_transfers_0_0010",
        "value": 0.001,
      },
      Object {
        "key": "global_settings_filter_spam_transfers_0_0100",
        "text": "global_settings_filter_spam_transfers_0_0100",
        "value": 0.01,
      },
      Object {
        "key": "global_settings_filter_spam_transfers_0_1000",
        "text": "global_settings_filter_spam_transfers_0_1000",
        "value": 0.1,
      },
      Object {
        "key": "global_settings_filter_spam_transfers_do_not",
        "text": "global_settings_filter_spam_transfers_do_not",
        "value": 0,
      },
    ]
  }
  renderLabel={[Function]}
  searchInput="text"
  selectOnBlur={true}
  selectOnNavigation={true}
  selection={false}
  value="default"
  wrapSelection={true}
/>
`;
